docs/design/verification_protocol.md

```diff
@@ -0,0 +1,122 @@
+# Public Health Authority Diagnosis Verification Protocol
+
+This design covers the exposure notifications server's ability to verify
+diagnosis certifications from public health authorities.
+
+The actual process of issuing these certificates is not covered in this
+document, but will be at a later date.
+
+## Motivation
+
+The current exposure notifications publication protocol only has the ability
+to authenticate that a request came from a mobile device via the vendor
+specific device attestation protocols. This doesn't give us the ability
+to certify that the diagnosis came from a trusted (by the server) public
+health authority in the jurisdiction.
+
+## Definitions
+
+* PHA : Public Health Authority
+* JWT : JSON Web Token
+* ECDSA : Elliptic Curve Digital Signature Algorithm
+* PIN : personal identification number. Could be a different authenticator.
+* HMAC : hash-based message authentication code
+* TEK : Temporary Exposure Key
+
+
+## High Level Flow
+
+1. App user is diagnosed with Covid-19. PHA issues a PIN code to the user.
+2. The user enters the PIN code on their App
+3. The app creates a HMAC value that combines the TEKs from the device along
+   with calculated transmission risk value, rolling period start and rolling
+	 period count values (the key data + metadata).
+	 These values are sent to the PHA server that issued the PIN.
+4. If the PIN is valid, the PHA issues a JWT that is signed using ECDSA over
+   the P-256 elliptic curve with SHA-256 as a hash function. The JWT includes
+   additional claims about the data (see below).
+5. The app on the user's device sends this signed JWT to the exposure
+   notifications server (this project).
+6. If the JWT is valid and signed by a trusted PHA (verified because the public
+	 key has been previously shared with the server), then the keys are imported
+	 into the server for distribution to other devices in the geography. Because
+   we are using pre-shared public keys, this verification can be done with
+   offline verification.
+
+## JWT Verification + Accepted Claims
+
+The exposure notifications server will accept this JWT in the current
+`verificationPayload` field in the exposure reporting API. When this is present,
+device attestations should be disabled. In addition, the exposure publishing
```

Comment:
Is 'should' correct? Or is it 'may'? If may be helpful to expand why? I agree both are not necessary.

Reply:
I think should is correct. I also think we should move the device verification into the reference implementation of the verification server.